static/panes/ir-view.js

```diff
@@ -85,17 +112,29 @@ Ir.prototype.initCallbacks = function () {
     this.eventHub.on('compiler', this.onCompiler, this);
     this.eventHub.on('compilerClose', this.onCompilerClose, this);
     this.eventHub.on('settingsChange', this.onSettingsChange, this);
+    this.eventHub.on('colours', this.onColours, this);
     this.eventHub.emit('irViewOpened', this._compilerid);
     this.eventHub.emit('requestSettings');
 
     this.container.on('resize', this.resize, this);
     this.container.on('shown', this.resize, this);
 };
 
-// TODO: de-dupe with compiler etc
+Ir.prototype.onColours = function (editorid, colours, scheme) {
+    if (editorid === this._editorid) {
+        var irColours = {};
```

Comment:
maybe use a let here to avoid it hoising out of scope.

Reply:
We can't use es6 features on client side code :(